src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotCatchGeneralExceptionTypesTests.cs

```diff
@@ -701,5 +701,70 @@ End Sub
                 }
             }.RunAsync();
         }
+
+        [Theory, WorkItem(3211, "https://github.com/dotnet/roslyn-analyzers/issues/3211")]
+        // No configuration - validate no diagnostics in default configuration
+        [InlineData("")]
+        // Check with exact method signature
+        [InlineData("dotnet_code_quality.CA1031.excluded_symbol_names = M:SomeNamespace.Test.M1(System.String)")]
+        [InlineData("dotnet_code_quality.excluded_symbol_names = M:SomeNamespace.Test.M1(System.String)")]
+        // Check with wildcard method signature
+        [InlineData("dotnet_code_quality.CA1031.excluded_symbol_names = M:SomeNamespace.Test.M*")]
+        [InlineData("dotnet_code_quality.excluded_symbol_names = M:SomeNamespace.Test.M*")]
+        // Check with wildcard type signature
+        [InlineData("dotnet_code_quality.CA1031.excluded_symbol_names = T:SomeNamespace.Tes*")]
```

Comment:
Interesting, not sure how the non-method tests are passing, given you check `excludedSymbols.Contains(method);`. Can you remind me if this option value leads to type symbol and all its member symbols being included in the resultant symbol set returned when option value is parsed?

Reply:
Ah, so I presume the `.Contains` works up the containing symbol chain. So, specifying a non-wildcard value like `[InlineData("dotnet_code_quality.CA1031.excluded_symbol_names = T:SomeNamespace.Test")]`, will cause the test to fail? We should ensure it is not such. I this also indicates that we are doing duplicate work in `ISymbolExtensions.IsConfiguredToSkipAnalysis`, that method has a loop walking up the containing symbol chain, and does a contains check for each symbol in this chain, which will in turn execute the wildcard based containing symbol chain walk.

Reply:
The wildcard walks-up the containing chain while the non-wildcard doesn't. In the example of `[InlineData("dotnet_code_quality.CA1031.excluded_symbol_names = T:SomeNamespace.Test")]` yes the test will fail (i.e. the diagnostic will still be reported). I'll check what's going-on when I use `IsConfiguredToSkipAnalysis` and get back to you.

Reply:
Yeah \o/ So using the `IsConfiguredToSkipAnalysis` helper allows this walk-up check! Now that I see that helper I think that my wildcard implementation is actually way to complex has I was already trying to do the walk-up chain while the helper already does a walk-up callback. I don't feel doing this now but I will leave a comment on the ticket to revisit the algo.